addons/sale_coupon/models/sale_order.py

```diff
@@ -154,10 +154,13 @@ def _get_reward_values_fixed_amount(self, program):
         # This allow manual overwrite of taxes for promotion.
         if program.discount_line_product_id.taxes_id:
             line_taxes = self.fiscal_position_id.map_tax(program.discount_line_product_id.taxes_id) if self.fiscal_position_id else program.discount_line_product_id.taxes_id
+            lines = self._get_paid_order_lines().filtered('price_reduce')
```

Comment:
`_get_paid_order_lines` will allow the order to go below 0 with fixed discounts in a case such as this one, I think we need to use `_get_base_order_lines(program)` but don't quote me on this 😅 :
- Product A; 10$
- 50% discount; -5$
- \>5$ fixed price discount; - \>5$ (possibly + tax)
We need to consider the